cmd/kubeadm/app/cmd/join.go

```diff
@@ -59,38 +59,40 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
 
 	var skipPreFlight bool
 	var cfgPath string
+	var discURL string
+	var discFile string
+	var discToken string
 
 	cmd := &cobra.Command{
 		Use:   "join <master address>",
 		Short: "Run this on any machine you wish to join an existing cluster",
 		Run: func(cmd *cobra.Command, args []string) {
-			j, err := NewJoin(cfgPath, args, &cfg, skipPreFlight)
+			j, err := NewJoin(cfgPath, discURL, discFile, discToken, args, &cfg, skipPreFlight)
 			kubeadmutil.CheckErr(err)
 			kubeadmutil.CheckErr(j.Validate())
 			kubeadmutil.CheckErr(j.Run(out))
 		},
 	}
 
 	cmd.PersistentFlags().StringVar(&cfgPath, "config", cfgPath, "Path to kubeadm config file")
+	cmd.PersistentFlags().StringVar(&discURL, "discovery-url", discURL, "Discovery URL")
+	cmd.PersistentFlags().StringVar(&discFile, "discovery-file", discFile, "Discovery File path")
+	cmd.PersistentFlags().StringVar(&discToken, "discovery-token", discToken, "Discovery Token")
+	cmd.PersistentFlags().StringVar(&discToken, "token", discToken, "This sets the token for both discovery and bootstrap auth")
```

Comment:
something like this should be added: `cmd.PersistentFlags().StringVar(&cfg.DiscoveryToken, "bootstrap-token", "", "blabla")`

then a defaulting/validation func should be created and be something like this:
```go
if len(token) != 0 && (len(bootstrapToken) != 0 || len(discoveryToken) != 0) {
    return fmt.Errorf("--token are mutually exclusive with --bootstrap-token and --discovery-token")
}
if len(token) == 0 && (len(bootstrapToken) == 0 || len(discoveryToken) == 0) {
    return fmt.Errorf("If --token is not specified, both --bootstrap-token and --discovery-token have to be provided")
}
if len(token) != 0 {
    bootstrapToken = token
    discoveryToken = token
}
```